Packs/CTIX/Integrations/CTIXv3/CTIXv3.py

```diff
@@ -464,93 +467,101 @@ def tag_indicator_updation(
         :param str object_type: Object type of the IOCs
         :param str object_id: Object ID of the IOCs
         :param dict params: Paramters to be added in request
         :return dict: Returns response for query
-        '''
+        """
         url_suffix = f"ingestion/threat-data/{object_type}/{object_id}/quick-actions/"
         client_url = self.base_url + url_suffix
         return self.get_http_request(client_url, **params)
 
     def get_indicator_relations(self, object_type: str, object_id: str, params: dict):
-        '''
+        """
         Get Indicator Relations
 
         :param str object_type: Object type of the IOCs
         :param str object_id: Object ID of the IOCs
         :param dict params: Paramters to be added in request
         :return dict: Returns response for query
-        '''
+        """
         url_suffix = f"ingestion/threat-data/{object_type}/{object_id}/relations/"
         client_url = self.base_url + url_suffix
         return self.get_http_request(client_url, **params)
 
     def get_indicator_observations(self, params: dict):
-        '''
+        """
         Get Indicator Observations
 
         :param dict params: Paramters to be added in request
         :return dict: Returns response for query
-        '''
+        """
         url_suffix = "ingestion/threat-data/source-references/"
         client_url = self.base_url + url_suffix
         return self.get_http_request(client_url, **params)
 
     def get_conversion_feed_source(self, params: dict):
-        '''
+        """
         Get Conversion Feed Source
 
         :param dict params: Paramters to be added in request
         :return dict: Returns response for query
-        '''
+        """
         url_suffix = "conversion/feed-sources/"
         client_url = self.base_url + url_suffix
         return self.get_http_request(client_url, **params)
 
     def get_lookup_threat_data(
-        self, object_type: str, object_names: list, params: dict
+        self, object_type: str, ioc_type: str, object_names: list, params: dict
     ):
-        '''
+        """
         Get Lookup Threat Data
 
         :param str object_type: Object type of the IOCs
         :param list object_names: Indicator/IOCs names
         :param dict params: Paramters to be added in request
         :return dict: Returns response for query
-        '''
+        """
         url_suffix = "ingestion/threat-data/list/"
+        query = f"type={object_type}"
+
+        if len(ioc_type) == 1:
+            query += f" AND ioc_type IN ('{ioc_type[0]}')"
+        elif len(ioc_type) > 1:
+            query += f" AND ioc_type IN {tuple(ioc_type)}"
+
         if len(object_names) == 1:
-            query = f"type={object_type} AND value IN ('{object_names[0]}')"
+            query += f" AND value IN ('{object_names[0]}')"
         else:
-            query = f"type={object_type} AND value IN {tuple(object_names)}"
+            query += f" AND value IN {tuple(object_names)}"
```

Comment:
If object_names is a list, the if condition is redundant as it is possible to just add tuple(the_list) like you do in the else.